File: services/backend/src/agent_cloud_backend/schemas/credential.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CredentialCreate(BaseModel):
    name: str
    base_url: str = ""
    api_key: str  # 明文,仅入站;绝不回显
    models: list[str] = []  # 该 provider 下可用模型清单(可多个)


class CredentialRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    name: str
    base_url: str
    masked: str
    models: list[str]
    created_at: datetime

File: apps/mobile/lib/models/credential.dart
/// BYOK provider 凭据(api_key 仅 masked 回传,明文不出后端)。
class ProviderCredential {
  final String id;
  final String name;
  final String baseUrl;
  final String masked; // 脱敏后的 key,如 sk-…1234
  final List<String> models;

  const ProviderCredential({
    required this.id,
    required this.name,
    required this.baseUrl,
    required this.masked,
    required this.models,
  });

  factory ProviderCredential.fromJson(Map<String, dynamic> j) =>
      ProviderCredential(
        id: j['id'] as String,
        name: j['name'] as String,
        baseUrl: j['base_url'] as String,
        masked: j['masked'] as String? ?? '',
        models: (j['models'] as List).cast<String>(),
      );
}
